src/classes/AffiliationMappingsHealthCheckVMapper.cls

```diff
@@ -577,6 +592,108 @@ public virtual with sharing class AffiliationMappingsHealthCheckVMapper {
         healthCheckGroupVModel.updateHealthCheckStatus();
         return healthCheckGroupVModel;
     }
+
+    /**********************************************************************************
+    * @description Retrieves a HealthCheckGroup view model representing the health check
+    * results for Account Record Types specified.  Determines whether the Account 
+    * record type included in an affiliation mapping is unique.
+    * @param  affiliationMappingsModel A model representing an affiliation mapping in EDA Settings
+    * @param  accountRecordTypeModel A model representing the Account record type specified for an affiliation mapping
+    * @param  duplicateAccountRecordTypeFieldValues A list of duplicate accountRecordTypeFieldValues retrieved from Affiliation Mappings
+    * @return A HealthCheckGroupVModel
+    **********************************************************************************/ 
+    @TestVisible
+    private virtual HealthCheckGroupVModel getAccountRecordTypeUniquenessHealthCheckGroupVModel(
```

Comment:
@xinnanlisf This will cover duplicates for a not found scenario. However, it is not covering duplicates when the same record type is used but is referenced by API name and the other is referenced by label. In order to achieve this, I recommend having the service return a data structure that contains the combination of duplicate record type models and duplicate record type labels. You will need a means to compare record type models, which you should be able to do based on the record type model's id.

Reply:
@MatthewBlanski good catch! Hesitating to add more complexity to the data model, does this work? I am adding the method to the vmapper layer instead of the service so that we can utilize those two maps.
`private Set<Id> getDuplicateAccountRecordTypeIds(
        List<AffiliationMappingsModel> affiliationMappingsModelsList,
        Map<String, RecordTypeModel> accountRecordTypeModelsByName,
        Map<String, RecordTypeModel> accountRecordTypeModelsByLabel 
    ) {
        Set<Id> accountRecordTypeIds = new Set<Id>();
        Set<Id> duplicateRecordTypeIds = new Set<Id>();

        for (AffiliationMappingsModel affiliationMappingsModel : affiliationMappingsModelsList) {
            RecordTypeModel accountRecordTypeModel = 
                this.locateRecordTypeService().getRecordTypeModelFromNameAndLabelMaps(
                    recordTypeNameOrLabel, 
                    accountRecordTypeModelsByName, 
                    accountRecordTypeModelsByLabel
                );
            
            if (accountRecordTypeModel == null) {
                continue;
            } else {
                if (accountRecordTypeIds.contains(accountRecordTypeModel.Id)) {
                    duplicateRecordTypeIds.add(accountRecordTypeModel.Id);
                } else {
                    accountRecordTypeIds.add(accountRecordTypeModel.Id);
                }
            }
        }

        return duplicateRecordTypeIds;
    }`

And then we check if duplicateRecordTypeIds.contains(accountRecordTypeModel.Id)

Reply:
That looks good to me

Reply:
ok on it!